esiclient/v1/node_baremetal.py

```diff
@@ -0,0 +1,61 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import logging
+
+import openstack
+from osc_lib.command import command
+from osc_lib.i18n import _
+
+
+class MDCBaremetalNodeList(command.Lister):
+    """List baremetal nodes from multiple OpenStack instances"""
+
+    log = logging.getLogger(__name__ + ".List")
+
+    def get_parser(self, prog_name):
+        parser = super(MDCBaremetalNodeList, self).get_parser(prog_name)
+        parser.add_argument(
+            '--clouds',
+            dest='clouds',
+            metavar='<clouds>',
+            nargs="+",
+            help=_("Specify the cloud to use from clouds.yaml.")
+        )
+
+        return parser
+
+    def take_action(self, parsed_args):
+
+        columns = ['Cloud', 'Region', 'UUID', 'Name', 'Instance UUID',
+                   'Power State', 'Provisioning State', 'Maintenance']
+        data = []
+
+        cloud_filter = parsed_args.clouds or []
+
+        cloud_regions = openstack.config.loader.OpenStackConfig().\
+            get_all_clouds()
+
+        for c in cloud_regions:
+
+            if cloud_filter and c._name not in cloud_filter:
```

Comment:
cloud_filter has a default value, so no need for the first part of the and clause

Reply:
its default value is to be empty if you dont pass in the --clouds flag. This means that nothing will print